src/post.rs

```diff
@@ -89,4 +89,29 @@ impl NewPost {
             .build()
             .expect("expected build to work")
     }
+
+    pub fn to_quote(&self, seckey: &[u8; 32], quoting: &Note) -> Note {
+        let new_content = format!(
+            "{}\nnostr:{}",
+            self.content,
+            enostr::NoteId::new(*quoting.id()).to_bech().unwrap()
+        );
+        let builder = NoteBuilder::new().kind(1).content(&new_content);
+
+        let builder = builder
+            .start_tag()
+            .tag_str("q")
+            .tag_str(&hex::encode(quoting.id()))
+            .sign(seckey);
+
+        let builder = builder
+            .start_tag()
+            .tag_str("p")
+            .tag_str(&hex::encode(quoting.pubkey()));
+
+        builder
+            .sign(seckey)
+            .build()
+            .expect("expected build to work")
```

Comment:
this should just be:

```suggestion
        let builder = NoteBuilder::new().kind(1).content(&new_content)
            .start_tag()
            .tag_str("q")
            .tag_str(&hex::encode(quoting.id()))
            .start_tag()
            .tag_str("p")
            .tag_str(&hex::encode(quoting.pubkey()))
            .sign(seckey)
            .build()
            .expect("expected build to work")
```